src/scheduler/FunctionCallClient.cpp

```diff
@@ -128,6 +140,22 @@ faabric::HostResources FunctionCallClient::getResources()
     return response;
 }
 
+void FunctionCallClient::sendAddPendingMigration(
+  std::shared_ptr<PendingMigrations> req)
+{
+    faabric::PendingMigrations request;
+    faabric::EmptyResponse response;
+
+    if (faabric::util::isMockMode()) {
+        faabric::util::UniqueLock lock(mockMutex);
+        addPendingMigrationRequests.emplace_back(host, req);
+    } else {
+        syncSend(faabric::scheduler::FunctionCalls::AddPendingMigration,
```

Comment:
Does this need to be synchronous? I think the sender will eventually wait on responses from the migrated `Executors` so could be async.

Reply:
The reason I made this synchronous was to guarantee that all executors see the pending migration at the same migration point. The problem I would want to guard against is:
1. Master host detects migration opportunity.
2. Executor hits `tryMigrate()` in the master host and it sees there is a pending migration.
3. Executor hits `tryMigrate()` in worker host but does not see a pending migration.
4. Worker host updates its migration map.
Note that 1 and 4 take a write lock on the scheduler, whereas 2 and 3 take a read one.

By making it synchronous we force 1 and 4 to happen atomically (with respect to 2 and 3).